crates/miden-objects/src/block/proposed_block.rs

```diff
@@ -330,17 +333,24 @@ fn check_nullifiers(
     for batch_nullifier in batch_nullifiers {
         match block_inputs.nullifier_witnesses().get(&batch_nullifier) {
             Some(witness) => {
-                let (_, nullifier_value) = witness
-                    .proof()
-                    .leaf()
-                    .entries()
-                    .iter()
-                    .find(|(key, _)| *key == batch_nullifier.inner())
-                    .ok_or(ProposedBlockError::NullifierProofMissing(batch_nullifier))?;
-
-                // Check if the nullifier is spent by comparing with EMPTY_WORD.
-                if *nullifier_value != EMPTY_WORD {
-                    return Err(ProposedBlockError::NullifierSpent(batch_nullifier));
+                match witness.proof().leaf() {
+                    // If the leaf is of variant Empty, the nullifier is unspent.
+                    SmtLeaf::Empty(_) => (),
+                    // If the leaf is of variant Single or Multiple, we have to check the value of
+                    // the entry.
+                    _ => {
+                        let (_, nullifier_value) = witness
+                            .proof()
+                            .leaf()
+                            .entries()
+                            .iter()
+                            .find(|(key, _)| *key == batch_nullifier.inner())
+                            .ok_or(ProposedBlockError::NullifierProofMissing(batch_nullifier))?;
+
+                        if *nullifier_value != EMPTY_WORD {
+                            return Err(ProposedBlockError::NullifierSpent(batch_nullifier));
+                        }
+                    },
```

Comment:
Could we note use `SmtProof::get()` method here? Something like:
```rust
if let Some(nullifier_value) = witness.proof().get(&batch_nullifier.inner()) {
    if nullifier_value != EMPTY_WORD {
        return Err(ProposedBlockError::NullifierSpent(batch_nullifier));
    }
}
```

Reply:
Ah thanks, I did not see that API.
Although we have to also map the case where `SmtProof::get` returns `None` to `NullifierProofMissing`, but I did that.